### ISSUE TITLE
Trouble importing into MEAD

### ISSUE THREAD

**Issue by dfarrell07** [dfarrell07/odl-ds-pack#5]
With a `vagrant up`, the Vagrantfile and build.sh script will now handle everything up to building ODL Parent, which fails with missing artifacts (as expected). Using the resulting logfile + RelEng import-from-logs script per the GDoc fails, as does importing directly by G-A. Note that I'm manually copying the log out of the Vagrant env and mounting the script on my dev box until #1 is fixed.

We first have many successful downloads from RH internal repos

```
[INFO] Downloading: http://download.eng.bos.redhat.com/brewroot/repos/rhos-10.0-odl-rhel-7-build/latest/maven/org/apache/mm
```

Then we try to reach an external repo, which rightly fails because we have /etc/hosts rules in place to block it.

```
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-deploy-plugin/2.8.2/maven-deploy-pp
```

The Vagrantfile handles copying the log out of the mock env (where we would use the RelEng scripts if we could mount, see #1). For now, I manually use the vagrant scp plugin to copy it out of the Vagrant box.

```
vagrant scp default:/vagrant/odl.odlparent.build.log odl.odlparent.build.log
```

Mount the RelEng scripts:

```
kinit <Kerberos ID?@REDHAT.COM
mkdir /mnt/redhat
sudo mount -o nolock vtap-bos-eng01.bos.redhat.com:/vol/engineering/devarchive/redhat /mnt/redhat
```

Just print the URLs to artifacts found by the script. Grepping against the results, the missing artifact isn't listed.

```
/mnt/redhat/scripts/rel-eng/utility/mead/mead-download-and-import --tag rhos-10.0-odl-rhel-7-maven-imports org.apache.maven.plugins:maven-deploy-plugin:2.8.2 | grep -i maven-deploy
# Returns nothing
```

Having the script both find the URLs a check if they are present, it reports that all 370 are already imported (which is true, since it's not including the missing one).

```
/mnt/redhat/scripts/rel-eng/utility/mead/mead-load-build-dependencies --print-missing ./odl.odlparent.build.log
```

Trying to import directly via artifact G-A-V fails:

```
/mnt/redhat/scripts/rel-eng/utility/mead/mead-download-and-import --tag rhos-10.0-odl-rhel-7-maven-imports org.apache.maven.plugins:maven-deploy-plugin:2.8.2
# prints "Error running command"
```

Also fails when using RH G value.

```
$ /mnt/redhat/scripts/rel-eng/utility/mead/mead-download-and-import --tag rhos-10.0-odl-rhel-7-maven-imports com.redhat.rcm.maven.plugin:maven-deploy-plugin:2.8.2
error running: /mnt/redhat/scripts/rel-eng/utility/mead/get-maven-artifacts com.redhat.rcm.maven.plugin:maven-deploy-plugin:2.8.2, return code was 1, output was:
Could not read http://repo1.maven.org/maven2/com/redhat/rcm/maven/plugin/maven-deploy-plugin/2.8.2/, continuing...
Could not read http://repository.jboss.org/nexus/content/groups/public-jboss/com/redhat/rcm/maven/plugin/maven-deploy-plugin/2.8.2/, continuing...
Could not read http://repository.jboss.org/maven2/com/redhat/rcm/maven/plugin/maven-deploy-plugin/2.8.2/, continuing...
Could not download artifacts for com.redhat.rcm.maven.plugin:maven-deploy-plugin:2.8.2
```

**Comment by mkolesni** [comment]
What does 'showmount -e vtap-bos-eng01.bos.redhat.com' give you?

**Comment by dfarrell07** [comment]
I don't have my laptop with me (weekend), but you should be able to
recreate the environment via Vagrant.

**Comment by dfarrell07** [comment]
> What does 'showmount -e vtap-bos-eng01.bos.redhat.com' give you?

@mkolesni did you mean this to be in the context of #1?

**Comment by dfarrell07** [comment]
```
$ sudo mount -o nolock vtap-bos-eng01.bos.redhat.com:/vol/engineering/devarchive/redhat /mnt/redhat
$ showmount -e vtap-bos-eng01.bos.redhat.com
Export list for vtap-bos-eng01.bos.redhat.com:
/vol/engineering/qa/rhts/lookaside/MW               download.lab.bos.redhat.com,10.16.0.0/16,10.18.0.0/15,file-test.bos.redhat.com,file.bos.redhat.com,jenkins.mw.lab.eng.bos.redhat.com,ns2.bos.redhat.com
/vol/engarchive_iso                                 rcm-jenkins-s01.app.eng.bos.redhat.com,rcm-dev.app.eng.bos.redhat.com,brewweb.devel.redhat.com,errata.devel.redhat.com,10.16.101.57,-x86-005.build.bos.redhat.com,-x86-006.build.bos.redhat.com,10.16.48.0/22,rcm-rhel7.app.eng.bos.redhat.com,10.16.101.121,10.16.101.122,10.16.101.123,10.16.101.124,10.19.48.0/22,rcm-rhel7-test.app.eng.bos.redhat.com
/vol/engineering/devarchive/redhat/devel/candidates errata.devel.redhat.com,armstrong.rhts.bos.redhat.com,apm-mustang-ev2-01.ml3.eng.bos.redhat.com,10.16.255.2,10.16.36.51,10.16.101.57,10.16.36.78,10.16.101.63,-x86-005.build.bos.redhat.com,-x86-006.build.bos.redhat.com,10.16.48.0/22,10.19.48.0/22,10.16.101.121,10.16.101.122,10.16.101.123,10.16.101.124,rcm-rhel7.app.eng.bos.redhat.com,10.16.101.126,10.16.101.75,rcm-rhel7-test.app.eng.bos.redhat.com,rcm-compose-01.host.stage.eng.bos.redhat.com,rcm-compose-nightly-01.host.prod.eng.bos.redhat.com,rcm-compose-nightly-02.host.prod.eng.bos.redhat.com,rcm-compose-release-01.host.prod.eng.bos.redhat.com,rcm-compose-legacy.host.prod.eng.bos.redhat.com
/vol/engineering/qa                                 download.lab.bos.redhat.com,10.16.0.0/16,10.18.0.0/15,armstrong.rhts.bos.redhat.com,file-test.bos.redhat.com,file.bos.redhat.com,ns2.bos.redhat.com
/vol/vol0/etc                                       netapp-admin,10.16.255.3
/vol/engineering                                    10.19.208.70,rcm-jenkins-s01.app.eng.bos.redhat.com,rcm-dev.app.eng.bos.redhat.com,pub.app.eng.bos.redhat.com,pkgwrangler.app.eng.bos.redhat.com,errata.devel.redhat.com,10.19.224.0/23,armstrong.rhts.bos.redhat.com,apm-mustang-ev2-01.ml3.eng.bos.redhat.com,10.16.101.82,10.16.255.2,10.16.36.51,10.16.101.57,10.16.36.78,10.16.101.63,-x86-005.build.bos.redhat.com,-x86-006.build.bos.redhat.com,10.19.48.0/22,10.16.48.0/22,10.16.50.142,10.16.50.143,10.16.50.144,10.16.101.121,10.16.101.122,10.16.101.123,10.16.101.124,rcm-rhel7.app.eng.bos.redhat.com,10.16.101.126,10.16.101.75,ppc-022.build.eng.bos.redhat.com,ppc-023.build.eng.bos.redhat.com,ppc-024.build.eng.bos.redhat.com,ppc-025.build.eng.bos.redhat.com,ppc-026.build.eng.bos.redhat.com,ppc-027.build.eng.bos.redhat.com,ppc-028.build.eng.bos.redhat.com,ppc-029.build.eng.bos.redhat.com,ppc-030.build.eng.bos.redhat.com,ppc-031.build.eng.bos.redhat.com,ppc-032.build.eng.bos.redhat.com,ppc-033.build.eng.bos.redhat.com,ppc-034.build.eng.bos.redhat.com,ppc-035.build.eng.bos.redhat.com,ppc-036.build.eng.bos.redhat.com,ppc-037.build.eng.bos.redhat.com,rcm-rawhide-02.app.eng.bos.redhat.com,rcm-rhel7-test.app.eng.bos.redhat.com,rcm-compose-01.host.stage.eng.bos.redhat.com,rcm-compose-nightly-01.host.prod.eng.bos.redhat.com,rcm-compose-nightly-02.host.prod.eng.bos.redhat.com,rcm-compose-release-01.host.prod.eng.bos.redhat.com,rcm-compose-legacy.host.prod.eng.bos.redhat.com
```

**Comment by dfarrell07** [comment]
The GDoc had recent updates about having network access for the MEAD import scripts (as Mike mentioned in email), but still said the build shouldn't have network access. I think this is wrong, and the problem. When I run the build with no `/etc/hosts` rules, it works (as expected) and the import scripts correctly parse the log to import missing artifacts (unexpected, per GDoc). I fixed the automated logic in 45808c95e287466530e2120c98f2f8cf8dcb4964 to make this work, added docs in 5e20f827a4242996ee7f48eeeb30f20e61cbe19e, removed the `/etc/hosts` config in the GDoc and clarified that the build should have network access, not just the MEAD scripts.
